src/main/java/org/jabref/model/texparser/TexParserResult.java

```diff
@@ -49,6 +51,13 @@ public TexParserResult() {
         return citations.get(key);
     }
 
+    /**
+     * Return a collection of citations using a BibEntry as reference.
+     */
+    public Collection<Citation> getCitationsByKey(BibEntry entry) {
+        return getCitationsByKey(Objects.requireNonNull(entry.getCiteKeyOptional().orElse(null)));
```

Comment:
better: `return entry.getCiteKeyOptional().map(this::getCitationsByKey).orElse(Collections.empty())`